src/server.ts

```diff
@@ -134,9 +134,14 @@ export const createApp = (
       proxyReq.removeHeader(SWARM_STAMP_HEADER)
       try {
         proxyReq.setHeader(SWARM_STAMP_HEADER, stampManager.postageStamp)
-      } catch (e) {
-        // There is no postage stamp to use
-        res.writeHead(503).end('Service Unavailable')
+      } catch (error) {
+        logger.error('proxy failure', error)
+
+        if (error && Reflect.get(error, 'message') === ERROR_NO_STAMP) {
+          res.writeHead(402).end(ERROR_NO_STAMP)
```

Comment:
I am not sure about this or maybe do not understand it.

When does this condition happen? To my understanding, this happens when users use gateway-proxy's stamp management and the stamp is not provided by the gateway-proxy? So maybe during the booting process? During the time when `/readiness` returns non-200? 

My only issue with this is that `402` is a client error, in this case, it would mean that the User should provide its own postage stamp (aka. the payment), but even if he would do so, it would not solve the issue as we do not pass on the stamp...

Reply:
I agree about the status code, it should not be `4xx`.

I think this can cover some edge cases, when a stamp runs out and a new one is being bought but the readiness check hasn't removed the pod yet... the gateway-watch can report confusing errors in that period, and this would clarify that